src/algebra/algebra/basic.lean

```diff
@@ -310,6 +310,21 @@ lemma map_eq_self (x : R) : algebra_map R R x = x := rfl
 
 end id
 
+section punit
+
+instance _root_.punit.algebra : algebra R punit :=
+{ to_fun := λ x, punit.star,
+  map_one' := rfl,
+  map_mul' := λ _ _, rfl,
+  map_zero' := rfl,
+  map_add' := λ _ _, rfl,
+  commutes' := λ _ _, rfl,
+  smul_def' := λ _ _, rfl }
```

Comment:
If you want:
```suggestion
by refine { to_fun := λ x, punit.star, smul := λ _ _, punit.star, ..}; intros; refl
```

Reply:
Will let the maintainer reviewing this tie break, I prefer my version but only mildly